pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go

```diff
@@ -191,6 +203,32 @@ func (pl *TaintToleration) ScoreExtensions() framework.ScoreExtensions {
 }
 
 // New initializes a new plugin and returns it.
-func New(_ context.Context, _ runtime.Object, h framework.Handle) (framework.Plugin, error) {
-	return &TaintToleration{handle: h}, nil
+func New(_ context.Context, _ runtime.Object, h framework.Handle, fts feature.Features) (framework.Plugin, error) {
+	return &TaintToleration{
+		handle:                    h,
+		enableSchedulingQueueHint: fts.EnableSchedulingQueueHint,
+	}, nil
+}
+
+// isSchedulableAfterPodChange is invoked whenever a pod changed. It checks whether
+// that change made a previously unschedulable pod schedulable.
+// When an unscheduled Pod, which was rejected by TaintToleration, is updated to have a new toleration,
+// it may make the Pod schedulable.
+func (pl *TaintToleration) isSchedulableAfterPodChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	originalPod, modifiedPod, err := util.As[*v1.Pod](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	if pod.UID == modifiedPod.UID &&
+		len(originalPod.Spec.Tolerations) != len(modifiedPod.Spec.Tolerations) {
```

Comment:
len(originalPod.Spec.Tolerations) != len(modifiedPod.Spec.Tolerations) =>

len(originalPod.Spec.Tolerations) < len(modifiedPod.Spec.Tolerations)

Reply:
This one should be fine. Tolerations are only allowed to add by api.